File: src/main/java/com/baedal/store/java/util/ObjectMapperUtil.java
package com.baedal.store.java.util;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;

public class ObjectMapperUtil {

	private static final ObjectMapper mapper = new ObjectMapper();

	static {
		mapper.registerModule(new JavaTimeModule());
	}

	public static String toJson(Object obj) {
		try {
			return mapper.writeValueAsString(obj);
		} catch (JsonProcessingException e) {
			throw new RuntimeException(e);
		}
	}
}

File: src/main/java/com/baedal/store/kotlin/util/ObjectMapperUtil.kt
package com.baedal.store.kotlin.util

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper

object ObjectMapperUtil {

    val mapper = ObjectMapper()

    fun toJson(raw: Any): String {
        val message = if (raw is String) JsonMessage(raw) else raw
        return mapper.writeValueAsString(message)
    }

    fun <T> fromJson(json: String, clazz: Class<T>): T {
        try {
            return mapper.readValue(json, clazz)
        } catch (e: JsonProcessingException) {
            throw RuntimeException(e)
        }
    }

    class JsonMessage(val message: String)
}
